src/Inputs/scene.inputManager.ts

```diff
@@ -899,6 +902,7 @@ export class InputManager {
         if (this._alreadyAttached) {
 
             this._deviceSourceManager.onInputChangedObservable.remove(this._onInputObserver);

```

Comment:
Would it be better to dispose the deviceSourceManager here and recreate on attachControl? This will avoid all the interface changes.

Reply:
I originally thought about that but I figured that recreating another DeviceSourceManager would mean that we'd lose any observers that a user created for connected, disconnected, and changed inputs.  I think that just affecting the events would be good enough.

Reply:
Hmm, interesting. I think you are right in principle. This does bring up an interesting problem with the way DeviceSourceManager is currently implemented. DeviceSourceManager is pointing to a singleton InternalDeviceSourceManager, but all the events are currently shared. If this continues, then one instance of DSM can unregister an event from another instance and vice versa. Similarly, dispose should probably dispose the events that have been register for a particular instance of DSM. Ugh, I'm not sure this DSM trick we did is a good idea anymore.

Reply:
Talked offline and we believe we don't need enable/disable events anymore and we can dispose the DSM now. This should simplify the code a bunch.

Reply:
enable/disable has been removed